scripts/ci-build.sh

```diff
@@ -478,21 +522,78 @@ MIRRORS_append = \"\\
 "
 fi
 
-# These environment variables control conditional compilation
-# of the SDK parts.
+# Deal with special setup, copy binary drivers etc.
 
-if [[ "$BUILD_SDK" != "true" ]]; then
-  bitbake pulsar-image
+# All R-Car targets start with "r-car-"
+set +e
+if echo "$TARGET" | egrep -q '^r-car-' ; then
+  if [[ -d "$BASEDIR/meta-renesas" ]] ; then
+    echo "Copying binary graphics and mmp drivers for $TARGET from: $SGX_GEN_3_DRIVERS"
+    cd "$BASEDIR/meta-renesas"
+    meta-rcar-gen3/docs/sample/copyscript/copy_evaproprietary_softwares.sh "$SGX_GEN_3_DRIVERS"
+    append_bblayers_conf meta-renesas 'BBLAYERS += " \
+      ${TOPDIR}/../meta-renesas/meta-rcar-gen3 \
+      ${TOPDIR}/../meta-ivi-renesas \
+      ${TOPDIR}/../meta-linaro/meta-optee \
+      "'
+  else
+    echo "TARGET is set to $TARGET but I did not find a BSP layer (meta-renesas)"
+    exit 2
+  fi
 fi
+set -e
 
-if [[ "$BUILD_TEST_IMAGE" == "true" ]]; then
-  # FIXME
-  bitbake pulsar-image-test
+case $TARGET in
+  r-car-m3-starter-kit)
+    target_local_conf_file=r-car-m3-starter-kit.local.conf
+    ;;
+
+  r-car-h3-starter-kit)
+    target_local_conf_file=r-car-h3-starter-kit.local.conf
+    ;;
+
+  r-car-m3-salvator-x)
+    target_local_conf_file=r-car-m3-salvator-x.local.conf
+    ;;
+
+  r-car-h3-salvator-x)
+    target_local_conf_file=r-car-h3-salvator-x.local.conf
+    ;;
+
+  *)
+    target_local_conf_file=
+    ;;
+esac
+
+
+if [ -n "$target_local_conf_file" ] ; then
+  # Download and use hardware-specific local conf settings (reused from the GDP project)
+  cd "$BASEDIR/build/conf"
+  local_conf_url="$GDP_TEMPLATES_URL/$target_local_conf_file"
+  fetch "$target_local_conf_file" "$local_conf_url"
```

Comment:
I wonder if this is the cause of your build failure. It seems the fetch will copy to build/conf. Line 581 appends "require r-car-m3-starter-kit.local.conf" to your local.conf. However if you actually look at r-car-m3-starter-kit.local.conf it [includes renesas-rcar-gen3.local.inc](https://github.com/GENIVI/genivi-dev-platform/blob/cc28fbfb130548f4cc9bf549844f2d5464a1bfef/gdp-src-build/conf/templates/r-car-m3-starter-kit.local.conf#L2) from the sub-directory template in GDP. Basically a difference in the build environment between meta-ivi and GDP. You have no template sub-directory in meta-ivi build/conf.

So I think at run-time in your environment it may not find the sub-directory so the majority of the required local.conf additions are not made. As the file uses an include, rather than require no error is thrown. A side effect of the missing local.conf is that [meta-linaro optee is not masked](https://github.com/GENIVI/genivi-dev-platform/blob/cc28fbfb130548f4cc9bf549844f2d5464a1bfef/gdp-src-build/conf/templates/renesas-rcar-gen3.local.inc#L23) which results in the CI failure you see.

That's based on a reading of the source. I don't have access to the CI build tree to check conclusively.

Reply:
I will check today but I'm quite sure you found it.   Now that you mention it I seem to remember having a similar thought, but never finished the thought it seems.  I pushed a fix, let's see

Reply:
My knowledge of the optee mask meant I could zero in quite quickly. Lucky in one regard that was the build failure :)

I almost just hacked the script to see if it fixed it but thought I could know for sure even quicker if I could access the build. So left it as you have that power.

Reply:
Oh I just remembered something else I noticed when I looked at the failed build last night. It was complaining about the layer compatibility not being set for meta-ivi-renesas. I addressed that so either it was just a timing thing (the build was before I pushed the change) or if SHAs are hard coded it (script?) needs an update.

Reply:
I just went back to consider the meta-ivi-renesas observation. You have this on meta-ivi master which is now moved to Thud. R-Car Thud community YBSP is not public yet. So this should move to sumo.

Reply:
OK let's try #99 instead